src/containers/Registration.tsx

```diff
@@ -0,0 +1,186 @@
+    )
+    .required("Введіть ваш пароль"),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref("password"), null], "Пароль має співпадати"),
+});
+
+const Registration = () => {
+  const [show, setShow] = useState(false);
+
+  const toggleShowState = () => setShow(!show);
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    reset,
+    formState: { errors },
+  } = useForm<MyForm>({
+    resolver: yupResolver(schema),
+  });
+
+  const submitForm = (data: MyForm) => {
+    console.log(data);
+    toggleShowState();
+    reset();
+  };
+
+  const watchUserName = watch("userName");
+
+  return (
+    <>
+      <StyledForm onSubmit={handleSubmit(submitForm)}>
+        <StyledFloatingLabel controlId="userNameInput" label="Ім'я">
+          <StyledFormControl
+            top="true"
+            {...register("userName")}
+            type="text"
+            name="userName"
+            placeholder="Ім'я"
+            maxLength={50}
+          />
+          <StyledCounterLabel>
+            {watchUserName ? watchUserName.length : 0} / 50
+          </StyledCounterLabel>
+        </StyledFloatingLabel>
+        <p className="text-danger">
+          {errors.userName && errors?.userName?.message}
+        </p>
+        <StyledFloatingLabel controlId="emailInput" label="Ел. пошта">
+          <StyledFormControl
+            {...register("email")}
+            type="text"
+            name="email"
+            placeholder="Ел. пошта"
+          />
+        </StyledFloatingLabel>
+        <p className="text-danger">{errors.email && errors?.email?.message}</p>
+        <StyledFloatingLabel
+          controlId="dateOfBirthInput"
+          label="Дата народження"
+        >
+          <StyledFormControl
+            {...register("dateOfBirth")}
+            type="date"
+            name="dateOfBirth"
+          />
+        </StyledFloatingLabel>
+        <p className="text-danger">
+          {errors.dateOfBirth && errors?.dateOfBirth?.message}
+        </p>
+        <StyledFloatingLabel controlId="passwordInput" label="Пароль">
+          <StyledFormControl
+            {...register("password")}
+            type="password"
+            name="password"
+            placeholder="Пароль"
+            autoComplete="on"
+          />
+        </StyledFloatingLabel>
+        <p className="text-danger">
+          {errors.password && errors?.password?.message}
+        </p>
+        <StyledFloatingLabel
+          controlId="confirmPasswordInput"
+          label="Підтвердіть пароль"
+        >
+          <StyledFormControl
+            {...register("confirmPassword")}
+            type="password"
+            name="confirmPassword"
+            placeholder="Підтвердіть пароль"
+            autoComplete="on"
+          />
+        </StyledFloatingLabel>
+        <p className="text-danger">
+          {errors.confirmPassword && errors?.confirmPassword?.message}
+        </p>
+        <div className="d-grid gap-2">
```

Comment:
interesting decision to use grid here, I can't agree with that, but anyway, you should use <Grid> component from bootstrap instead

Reply:
![image](https://user-images.githubusercontent.com/103769012/185008788-4f69ccc3-56e5-4261-895a-712939355644.png)
This construct is from react bootstrap documentation :)

Reply:
Now using Bootstrap component instead styled component